src/analyzer/protocol/ssh/ssh-analyzer.pac

```diff
@@ -120,6 +120,16 @@ refine flow SSH_Flow += {
 		return true;
 		%}
 
+	function proc_ssh2_keys_setup_direction(is_orig: bool): bool
```

Comment:
Thanks for the PR!

(I hope someone else from the team with more experience chimes in) I see the convenience of the generic "setup direction" event, but do wonder if introducing 4 new (low-level) events for the individual `_INIT` and `_SECRET` messages would be the more usual approach and provide interesting additional information for other scenarios (and also make it easier to debug/localize in case things don't quite work as expected).

Minimally they all could provide `c: connection` and `is_orig: bool` and possibly additional parameters if available in the message (seeing e, output_token, encrypted_payload). But adding more parameter is always possible without breaking backwards compat.

It certainly does add more boilerplate in the .pac files and also in scripts they'd all need to be handled rather than having one specific event everything is funneled through though.

Reply:
I like this suggestion as well — it shifts the eventing more toward mechanism, gaining flexibility down the road.